docs/reference/configuration/Storage.md

```diff
@@ -94,3 +96,97 @@ Name: nvstore.max_keys
     Macro name: NVSTORE_MAX_KEYS
     Value: 16 (set by library:nvstore)
 ```
+
+### BlockDevice - default configuration
+
+The Mbed OS configuration allows you to add block devices as components using the `targets.json` file or target overrides in the application configuration file.
+
+When one of the following components is enabled, a default block device is set in the system:
+
+1. SPIF component.
+1. DATAFLASH component.
+1. SD component.
+
+Components can coexist in the system. A device can have SPIF and SD or any combination of block devices enabled but only one default block device.
+
+The list above is in the order of precedence shows which block device is the default one if more than one component is enabled.
+
+#### Configuring components
+
+Adding "components": ["???"] in targets.json:
```

Comment:
@theotherjimmy As it's the weekend in Israel, could you please tell me what should go here?

Reply:
I think that's going to be one of "SD", "SPIF" or "FLASHIAP"

Reply:
This should just say "for example, the following entry in `targets.json` enables the SD component". The following example should be trimmed to only identifying info and the important information.

Reply:
```
 "K64F": {
        "components": ["SD"],
        "core": "Cortex-M4F",
        "supported_toolchains": ["ARM", "GCC_ARM", "IAR"],
        "inherits": ["Target"],
        "features": ["STORAGE"],
        "release_versions": ["2", "5"],
        ...
    },
```

Is probably enough to get the point across.